### PR TITLE
JBPM-9979: Mark async active nodes and change default active node border

### PR DESCRIPTION
**Pull request by nmirasch**
Jira: https://issues.redhat.com/browse/JBPM-9979  

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by nmirasch** [comment]
@bxf12315 I've created this PR based in your commit related with the errorBorderColor in order to:
- Keep backward compatibility just adding new calls to allow to use the new color option.
- Replace the references to errorBorderColor by asyncActiveborderColor.
- I've replaced the test by something we can found in real execution to try the fullLogs logic with a real case

Could you take a look on this one to see if there is something missing here?

**Comment by bxf12315** [comment]
> @bxf12315 I've created this PR based in your commit related with the errorBorderColor in order to:
> 
> * Keep backward compatibility just adding new calls to allow to use the new color option.
I do not know why we need to create a new call? the reset call's any params could be null.  the rest API just using one places.  

> * Replace the references to errorBorderColor by asyncActiveborderColor.
Maybe failedAsyncNodeBoraderColor is better.
> * I've replaced the test by something we can found in real execution to try the fullLogs logic with a real case
> 
> Could you take a look on this one to see if there is something missing here?

**Comment by nmirasch** [comment]
@bxf12315 Thanks,
* the thing is to avoid to change the current existing calls introducing that needed null. Just adding a new call if it needs to used the new border color conf, in order to avoid break in case someone is using the old one..
* failedAsyncNodeBorderColor implies it's failing but it's not completely true. They are async nodes that not have been completed yet.. At this point we can't differentiate between in 'error' or 'queued' ones.. that's the reason to use 'active' instead

**Comment by nmirasch** [comment]
Jenkins retest this

**Comment by bxf12315** [comment]
> * oid to change the current existing calls introducing t



> @bxf12315 Thanks,
> 
> * the thing is to avoid to change the current existing calls introducing that needed null. Just adding a new call if it needs to used the new border color conf, in order to avoid break in case someone is using the old one..
> * failedAsyncNodeBorderColor implies it's failing but it's not completely true. They are async nodes that not have been completed yet.. At this point we can't differentiate between in 'error' or 'queued' ones.. that's the reason to use 'active' instead

The two rest API has the same URL path, just has different query param and it's a GET. Maybe you new param is not work.
and I think it do not break other exits call, if the param is null, the the call also work.
Do you try the code, the async nodes is work?

**Comment by nmirasch** [comment]
@bxf12315  yes, I agree we can keep just one rest endpoint.(I'm going to add that change now)  The issue was regarding the api changes method java.lang.String org.kie.server.client.UIServicesClient::getProcessInstanceImageCustomColor with different number or parameters could make mandatory to introduce changes in older calls, to use kie-server with this change.

**Comment by bxf12315** [comment]
@nmirasch the code also not work. QueryParam's ASYNC_ACTIVE_BORDER_COLOR, will not work. you need to use "SVG_NODE_ASYNC_ACTIVE_BORDER_COLOR"

**Review by cristianonicolai — APPROVED** [review]

File: kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/UIServicesClient.java [hunk 1]
```diff
@@ -188,6 +188,20 @@
...
 
+    /**
+     * Returns process image (svg) with annotated active and completed nodes for given process instance
+     * that belongs to given container
+     * @param containerId container identifier where process resides
+     * @param processInstanceId unique process instance id
+     * @param completeNodeColor process instance image complete node color
+     * @param completeNodeBorderColor process instance image complete node border color
+     * @param activeNodeBorderColor process instance image active node border color
+     * @param activeAsyncNodeBorderColor process instance image error node border color
```
Comment (barboras7) [inline kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/UIServicesClient.java:199]:
```suggestion
     * @param activeAsyncNodeBorderColor process instance image active async node border color
```

**Review by barboras7 — CHANGES_REQUESTED** [review]

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Comment by bxf12315** [comment]
> jenkins please retest this

Please don't merge this, it's a duplicate PR.

**Review by barboras7 — APPROVED** [review]

**Pull request closed without merging**